tests/drivers/gpio/gpio_ite_it8xxx2_v2/src/main.c

```diff
@@ -0,0 +1,462 @@
+/*
+ * Copyright 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/dt-bindings/gpio/ite-it8xxx2-gpio.h>
+#include <zephyr/fff.h>
+#include <zephyr/ztest.h>
+
+#define MY_SERIAL DT_NODELABEL(gpioa)
+
+const struct device *const gpio_dev = DEVICE_DT_GET(MY_SERIAL);
+
+/* These values must match what is set in the dts overlay. */
+#define TEST_PIN  1
+#define TEST_IRQ  2
+#define TEST_MASK BIT(4)
+
+DEFINE_FFF_GLOBALS;
+
+uint8_t ite_intc_get_irq_num(void)
+{
+	return posix_get_current_irq();
+}
+
+static uint8_t reg_fake;
+static uint8_t reg_gpdmr;
+static uint8_t reg_gpdr;
+static uint8_t reg_gpotr;
+static uint8_t reg_p18scr;
+static uint8_t reg_wuemr, reg_wuesr, reg_wubemr;
+static uint8_t reg_gpcr[8];
+static bool clear_gpcr_before_read;
+
+unsigned int *fake_ecreg(intptr_t r)
+{
+	switch (r) {
+	case 0x00f01601: /* GPDR */
+		return (unsigned int *)&reg_gpdr;
+	case 0x00f01618: /* GPDMR */
+		return (unsigned int *)&reg_gpdmr;
+	case 0x00f01630: /* GPOTR */
+		return (unsigned int *)&reg_gpotr;
+	case 0x00f01648: /* P18SCR */
+		return (unsigned int *)&reg_p18scr;
+	case 0xf01b20:
+		return (unsigned int *)&reg_wuemr;
+	case 0xf01b21:
+		return (unsigned int *)&reg_wuesr;
+	case 0xf01b23:
+		return (unsigned int *)&reg_wubemr;
+	}
+	if (r >= 0x00f01660 && r < 0x00f01668) {
+		if (clear_gpcr_before_read) {
+			reg_gpcr[r - 0x00f01660] = 0;
+		}
+		return (unsigned int *)&reg_gpcr[r - 0x00f01660];
+	}
+	zassert_unreachable("Register access: %x", r);
+	return (unsigned int *)&reg_fake;
+}
+
+static int callback_called;
+
+static void callback(const struct device *port, struct gpio_callback *cb, gpio_port_pins_t pins)
+{
+	callback_called++;
+	zexpect_equal(pins, BIT(TEST_PIN));
+}
+
+static struct gpio_callback callback_struct;
+
+static void before_test(void *fixture)
+{
+	callback_called = 0;
+	reg_gpdmr = 0;
+	reg_gpdr = 0;
+	reg_gpotr = 0;
+	reg_p18scr = 0;
+	reg_wuemr = 0;
+	reg_wuesr = 0;
+	reg_wubemr = 0;
+	memset(reg_gpcr, 0, sizeof(reg_gpcr));
+	clear_gpcr_before_read = false;
+}
+
+static void after_test(void *fixture)
+{
+	if (callback_struct.handler != NULL) {
+		zassert_ok(gpio_remove_callback(gpio_dev, &callback_struct));
+	}
+	callback_struct.handler = NULL;
+}
+
+ZTEST_SUITE(gpio_ite_it8xxx2_v2, NULL, NULL, before_test, after_test, NULL);
+
+ZTEST(gpio_ite_it8xxx2_v2, test_get_active_high)
+{
+	zassert_true(device_is_ready(gpio_dev));
+	zassert_ok(gpio_pin_configure(gpio_dev, TEST_PIN, GPIO_INPUT | GPIO_ACTIVE_HIGH));
+	zexpect_equal(reg_gpotr, 0, "reg_gpotr=%x", reg_gpotr);
+	zexpect_equal(reg_p18scr, 0);
+	zexpect_equal(reg_gpcr[TEST_PIN], GPCR_PORT_PIN_MODE_INPUT, "reg_gpcr[%d]=%x", TEST_PIN,
+		      reg_gpcr[TEST_PIN]);
+	reg_gpdmr = (uint8_t)~BIT(TEST_PIN);
+	zassert_false(gpio_pin_get(gpio_dev, 1));
```

Comment:
Use TEST_PIN consistently.
```suggestion
	zassert_false(gpio_pin_get(gpio_dev, TEST_PIN));
```

Reply:
done